includes/Checker/Abstract_Check_Runner.php

```diff
@@ -330,7 +375,16 @@ final public function get_checks_to_run() {
 			$check_flags = $check_flags | Check_Repository::INCLUDE_EXPERIMENTAL;
 		}
 
-		return $this->check_repository->get_checks( $check_flags, $check_slugs );
+		$checks = $this->check_repository->get_checks( $check_flags, $check_slugs );
+
+		// Filters the checks by specific categories.
+		$get_categories = $this->get_categories();
+		if ( $get_categories ) {
+			$check_categories = new Check_Categories();
+			$checks           = $check_categories->filter_checks_by_categories( $checks, $get_categories );
```

Comment:
A bit of an odd variable name, why not just `$categories`? Also, similar to the above, no need to instantiate the class when the methods are static.
```suggestion
		$categories = $this->get_categories();
		if ( $categories ) {
			$checks = Check_Categories::filter_checks_by_categories( $checks, $categories );
```